tests/suites/test_suite_psa_crypto.data

```diff
@@ -3029,9 +3029,9 @@ PSA AEAD decrypt: ChaCha20-Poly1305 (good tag, zero-length input)
 depends_on:PSA_WANT_ALG_CHACHA20_POLY1305:PSA_WANT_KEY_TYPE_CHACHA20
 aead_decrypt:PSA_KEY_TYPE_CHACHA20:"808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9f":PSA_ALG_CHACHA20_POLY1305:"070000004041424344454647":"":"a0784d7a4716f3feb4f64e7f4b39bf04":"":PSA_SUCCESS
 
-PSA AEAD decrypt: ChaCha20-Poly1305 (nonce=8, not supported)
+PSA AEAD decrypt: ChaCha20-Poly1305 (nonce=8, not supported in single operation)
 depends_on:PSA_WANT_ALG_CHACHA20_POLY1305:PSA_WANT_KEY_TYPE_CHACHA20
-aead_decrypt:PSA_KEY_TYPE_CHACHA20:"808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9f":PSA_ALG_CHACHA20_POLY1305:"0700000040414243":"":"a0784d7a4716f3feb4f64e7f4b39bf04":"":PSA_ERROR_NOT_SUPPORTED
+aead_decrypt:PSA_KEY_TYPE_CHACHA20:"808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9f":PSA_ALG_CHACHA20_POLY1305:"0700000040414243":"":"a0784d7a4716f3feb4f64e7f4b39bf04":"":PSA_ERROR_INVALID_SIGNATURE
```

Comment:
Can you explain the updated expected error code? `PSA_ERROR_NOT_SUPPORTED` still sounds correct to me.

Reply:
Because earlier 8 byte nonces were rejected by the `psa_aead_check_nonce_length` function but now that is accepted. However the single operation function assumes that the nonce length is 12 bytes and hence the operation is carried out but the result is wrong. 
In the definition of `psa_aead_decrypt`  PSA_ERROR_INVALID_SIGNATURE is interpreted as the ciphertext is not authentic.

Reply:
It sounds like we are adapting the test in order to fit our implementation, when we really should be treating the test as a specification for what the implementation should do. `PSA_ERROR_NOT_SUPPORTED ` is more correct/informative than `PSA_ERROR_INVALID_SIGNATURE ` (the signature might actually be correct). Since we're passing the nonce length as an argument, it should be simple to check that we're actually getting a 12 byte nonce (which is the only length supported through the single-shot implementation) before/after passing it to `psa_aead_check_nonce_length()`. Or perhaps it's even better update the helper function in order to account for the differences in support between single-shot and multi-part operations.